thunder/torch/__init__.py

```diff
@@ -2844,6 +2844,39 @@ def _normalize(a: TensorProxy, /, norm_dims, eps: Number) -> tuple[TensorLike, T
     return out, mean, rstd
 
 
+@torchsymbol(torch.nn.functional.normalize, is_method=True)
+def normalize(
+    a: TensorProxy, /, p: float = 2.0, dim: int | Sequence[int] = 1, eps: float = 1e-12, out: None | TensorProxy = None
+) -> TensorProxy:
+    utils.check(
+        dtypes.is_float_dtype(a.dtype) or dtypes.is_complex_dtype(a.dtype),
+        lambda: f"normalize: Expected a floating point or complext tensor as input. Got {a.dtype}",
+        TypeError,
+    )
+    utils.check(out is None, lambda: "normalize: out is not None which is currently unsupported", NotImplementedError)
+    computation_dtype, result_dtype = _reduction_dtypes(a, REDUCTION_OUTPUT_TYPE_KIND.COMPLEX_TO_FLOAT, a.dtype)
```

Comment:
Is this really complex to float? I thought normalizing a complex tensor also produced a complex tensor?

Reply:
Yeah, normalizing a complex tensor produces a complex tensor. And the current implementation of thunder.normalize also outputs a complex due to:
```
out = a / denom
```
Most of the implementation was inspired by `vector_norm` from PyTorch which I expect should be implemented in Thunder. However, I can get rid of this for now as it does not have much effect. What do you think?

Reply:
Eventually to handle different `dim` cases, I think I should implement `vector_norm` and `matrix_norm` like PyTorch. I can do this in the next PR if it is an acceptable approach!

Reply:
`vector_norm` and `matrix_norm` would be great to implement! I see what you mean now about the dtype stuff with complex. No need to change it!